firmware-support/bittide-sys/src/stability_detector.rs

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2025 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use bittide_hal::{
+    manual_additions::timer::{Duration, Instant},
+    shared::devices::clock_control::ClockControl,
+};
+use itertools::izip;
+use ufmt::derive::uDebug;
+
+pub struct StabilityDetector {
+    data_counts: [i32; ClockControl::DATA_COUNTS_LEN],
+    starts: [Option<Instant>; ClockControl::DATA_COUNTS_LEN],
```

Comment:
Would it make sense to make this a const generic?

```suggestion
pub struct StabilityDetector<const DATA_COUNTS_LEN: usize> {
    data_counts: [i32; DATA_COUNTS_LEN],
    starts: [Option<Instant>; DATA_COUNTS_LEN],
```